zokrates_core/src/helpers/wasm.rs

```diff
@@ -0,0 +1,385 @@
+use std::fmt;
+use std::fs::File;
+use helpers::{Signed, Executable};
+use field::Field;
+use std::panic;
+
+use wasmi::{Module, ModuleInstance, ImportsBuilder, NopExternals, ModuleRef};
+
+use rustc_hex::FromHex;
+
+#[derive(Clone, PartialEq, Debug, Serialize, Deserialize)]
+pub enum WasmHelper {
+	File(String),
+	Code(String),
+}
+
+impl WasmHelper {
+	/* Hand-coded assembly for identity */
+	pub const IDENTITY_WASM : &'static str = "0061736d010000000105016000017f030302000005030100010615047f0041010b7f0041010b7f0041200b7f0141000b074b06066d656d6f727902000e6765745f696e707574735f6f6666000105736f6c766500000a6d696e5f696e7075747303000b6d696e5f6f75747075747303010a6669656c645f73697a6503020a2c0225000340412023036a410023036a280200360200230341016a240323032302470d000b41200b040041000b0b4b020041000b20ffffffff000000000000000000000000ffffffff0000000000000000000000000041200b20deadbeef000000000000000000000000deadbeef000000000000000000000000";
+	/* Generated from C code, normalized and cleaned up by hand */
+	pub const BITS_WASM : &'static str = "0061736d01000000010c026000017f60037f7f7f017f0304030000010405017001010105030100020626067f0141f0c7040b7f0041f0c7040b7f0041f0c7000b7f0041200b7f0041010b7f0041fe010b074b06066d656d6f727902000e6765745f696e707574735f6f6666000005736f6c766500010a6669656c645f73697a6503030a6d696e5f696e7075747303040b6d696e5f6f75747075747303050901000ad5030305004190080b4f01027f41b008210041b008410041c03f10021a4100210103402000410120014107717420014103764190086a2d0000714100473a0000200041206a2100200141016a220141fe01470d000b41b0080bfc0202037f017e02402002450d00200020013a0000200020026a2203417f6a20013a000020024103490d00200020013a0002200020013a00012003417d6a20013a00002003417e6a20013a000020024107490d00200020013a00032003417c6a20013a000020024109490d002000410020006b41037122046a2203200141ff017141818284086c22013602002003200220046b417c7122046a2202417c6a200136020020044109490d002003200136020820032001360204200241786a2001360200200241746a200136020020044119490d00200320013602102003200136020c2003200136021420032001360218200241686a2001360200200241646a20013602002002416c6a2001360200200241706a20013602002004200341047141187222056b22024120490d002001ad22064220862006842106200320056a2101034020012006370300200141086a2006370300200141106a2006370300200141186a2006370300200141206a2101200241606a2202411f4b0d000b0b20000b";
+
+	pub fn new_file<U: Into<String>>(u: U) -> Self {
```

Comment:
Maybe implement `From<PathBuf>` for WasmHelper

Reply:
I currently have `From<&str>` and `From<&mut file>` which I'm told by the compiler could be turned into one thing if I enable the proper nightly feature. Still investigating this one.

Reply:
So we're trying to keep zokrates_core free of fs operations, so I anticipate the constructor would take the bytecode here, it would have been fetched in a previous layer.